pkg/apis/garden/validation/validation.go

```diff
@@ -1424,12 +1455,15 @@ func validateKubernetesVersionUpdate(new, old string, fldPath *field.Path) field
 
 func validateDNS(dns garden.DNS, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-
-	if dns.Provider != garden.DNSUnmanaged && dns.Provider != garden.DNSAWSRoute53 && dns.Provider != garden.DNSGoogleCloudDNS && dns.Provider != garden.DNSOpenstackDesignate {
-		allErrs = append(allErrs, field.NotSupported(fldPath.Child("provider"), dns.Provider, []string{string(garden.DNSUnmanaged), string(garden.DNSAWSRoute53), string(garden.DNSGoogleCloudDNS), string(garden.DNSOpenstackDesignate)}))
+	if !availableDNS.checkProvider(dns.Provider) {
+		allErrs = append(allErrs, field.NotSupported(fldPath.Child("provider"), dns.Provider, availableDNS.list()))
 	}
 
-	if dns.HostedZoneID != nil {
+	if dns.Provider == garden.DNSAlicloud {
+		if dns.HostedZoneID != nil {
```

Comment:
This conflicts with Shoot reconciliation when Shoot is set up with a customer's domain. https://github.com/gardener/gardener/blob/cb61e3ca143283691f98a8c5675e9852c1714cda/pkg/operation/botanist/dns.go#L324

Reply:
@jia-jerry what is the reason for not allowing a hosted zone id in Alicloud DNS?

Reply:
@rfranzke There is **NO** Hosted Zone information.
1. Not found in in terraform resource structure: https://www.terraform.io/docs/providers/alicloud/d/dns_records.html
2. Not found in portal page:
https://dns.console.aliyun.com
3. Not found in API:
https://help.aliyun.com/document_detail/29772.html (In Chinese)

Reply:
@timuthy  This change does not break other DNS providers validation. Would you please check it ?

Reply:
@timuthy 
Thank you for your suggestion. Will fix the issue.
@rfranzke 
I found to add a DNS provider, it needs to take care of many switch cases statements.

Is there a plan to do some refactory? Just make it like a cloud botainist?

Reply:
> @rfranzke There is NO Hosted Zone information.

OK, thank you.

> Is there a plan to do some refactory? Just make it like a cloud botainist?

Definitely, see GEP proposal (open PR).